app/charges/controllers.py

```diff
@@ -269,6 +275,52 @@ def edit_charge(user, user_data):
             key == "status" or key == "paw_links" or key == "private" or key == "committee"):
             setattr(charge, key, user_data[key])
 
+        # Edit progress notes
+        if (key == "progress_notes"): 
+            current_notes = getattr(charge, "progress_notes") 
+
+            # Get the progress note info from user_data
+            note_id = int(user_data["progress_notes"]["id"])
+            note = str(user_data["progress_notes"]["note"])
+            date = str(user_data["progress_notes"]["date"])
+
+            # Edit the notes, and update the DB
+            current_notes.edit(note_id, [note, date, str(note_id)])
+            setattr(charge, "progress_notes", current_notes) 
+
+        #Delete progress_notes
+        if (key == "delete_note"): 
+            current_notes = getattr(charge, "progress_notes") 
+
+            note_id = int(user_data["delete_note"]["id"])
+
+            current_notes.pop(note_id)
+            setattr(charge, "progress_notes", current_notes)
+
+        # Create a new progess_note
+        if (key == 'add_note'):
+            # Get the current list of notes
+            current_notes = getattr(charge, "progress_notes")
+            if (current_notes == None):
+                current_notes = []
+
+            # Get the progress note info from user_data
+            note_id = str(user_data["add_note"]["id"])
+            note = str(user_data["add_note"]["note"])
+            date = str(user_data["add_note"]["date"])
+
+            # Create the new one from the user data
+            new_note = []
+            new_note.append(note)
+            new_note.append(date)
+            new_note.append(note_id)
```

Comment:
Is there a way to save the note as an object and not an array of strings?

Reply:
After looking into it, it would be possible to use jsonb objects (https://www.postgresql.org/docs/9.5/datatype-json.html) However the functionality wouldn't change at all and it would be a significant refactoring of the branch. I'm not sure its worth the development time at this point.